API V2: /storage-providers endpoint

jcace opened the pull request:
Remove the `/public/miners` group of endpoints and work them into `/storage-providers` endpoints.

Closes #578

jcace commented:
questions:

 I notice we have a lot of duplicate endpoints calling the same handlers - is the intention to remove them?
(for example `/deals/query/:miner` and the new `/storage-providers/:sp/queryask` both use the `handleSpQueryAsk` handler.

Also, are we planning to change the `/admin/miners` endpoints as part of this change?

neelvirdy commented on handlers.go:
```diff
@@ -237,12 +237,12 @@ func (s *Server) ServeAPI() error {
-	miners.GET("/deals/:miner", s.handleGetMinerDeals)
-	miners.GET("/stats/:miner", s.handleGetMinerStats)
-	miners.GET("/storage/query/:miner", s.handleQueryAsk)
+	storageProviders := e.Group("/storage-providers")
```
can we configure a reroute so clients that call the old /miners endpoints get rerouted to the /storage-providers endpoints with their request intact?

jcace replied:
@gmelodie is the intent to have backwards compatibility with v2->v0 api endpoints? I thought I saw somewhere that it would be totally disparate.

gmelodie replied:
@neelvirdy we will keep old endpoints the way they are at least for a good while. It's okay to delete them in this PR but we'll have to rebase and put them back in after https://github.com/application-research/estuary/pull/596 is merged. 

Short answer: yes break everything

neelvirdy replied:
ok gotcha, https://github.com/application-research/estuary/pull/596 is what i was missing. it makes more sense to me to merge #596 before starting the rest of this work so you can incrementally contribute these to `dev` and avoid rebasing conflicts, but defer to you

neelvirdy commented on handlers.go:
```diff
@@ -185,7 +185,7 @@ func (s *Server) ServeAPI() error {
 	deals.GET("/status/:deal", withUser(s.handleGetDealStatus))
 	deals.GET("/status-by-proposal/:propcid", withUser(s.handleGetDealStatusByPropCid))
-	deals.GET("/query/:miner", s.handleQueryAsk)
+	deals.GET("/query/:miner", s.handleSpQueryAsk)
```
nit `:miner` -> `:sp` or `:storage_provider` (also on line below)

neelvirdy commented on handlers.go:
```diff
@@ -237,12 +237,12 @@ func (s *Server) ServeAPI() error {
-	miners.GET("/storage/query/:miner", s.handleQueryAsk)
+	storageProviders := e.Group("/storage-providers")
+	storageProviders.GET("", s.handleAdminGetStorageProviders)
+	storageProviders.GET("/:sp/deals", s.handleSpGetDeals)
```
nit: i prefer being verbose so `handleSpGetDeals` -> `handleStorageProviderGetDeals`, if no one disagrees. same with the other handlers here

neelvirdy commented on handlers.go:
```diff
@@ -2100,31 +2099,31 @@ func (s *Server) handleGetSystemConfig(c echo.Context, u *util.User) error {
 }
 
-type minerResp struct {
+type spResp struct {
```
nit: same verbose nit on these sp structs

jcace commented:
@neelvirdy addressed your comments!

There are a bunch of other places that have `miners` throughout the endpoints - may be worth it to create a separate task to go through and change them everywhere - though, currently I'm not 100% sure about what we're doing with those other endpoints. Need a bit more clarity

gmelodie reviewed (commented):
Looks good. But I don't see where you removed the `/public` endpoints as per #578

jcace commented:
> Looks good. But I don't see where you removed the /public endpoints as per https://github.com/application-research/estuary/issues/578

hey @gmelodie  what do you mean? `/public/miners` should be completely removed. We're not handling the other `/public` endpoints in this PR are we?

jcace commented:
closing in favour of #818
